p2p/src/authenticated/actors/peer/actor.rs

```diff
@@ -124,55 +157,25 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng> Actor<E> {
-                                    let msg = wire::Message{
-                                        payload: Some(wire::message::Payload::BitVec(bit_vec)),
-                                    }.encode_to_vec();
-                                    conn_sender.send(&msg)
-                                        .await
-                                        .map_err(Error::SendFailed)?;
-                                    self.sent_messages
-                                        .get_or_create(&metrics::Message::new_bit_vec(&peer))
-                                        .inc();
-                                }
-                                Message::Peers { peers: msg } => {
-                                    let msg = wire::Message{
-                                        payload: Some(wire::message::Payload::Peers(msg)),
-                                    }.encode_to_vec();
-                                    conn_sender.send(&msg)
-                                        .await
-                                        .map_err(Error::SendFailed)?;
-                                    self.sent_messages
-                                        .get_or_create(&metrics::Message::new_peers(&peer))
-                                        .inc();
-                                }
+                            let (payload, metric_msg) = match msg {
+                                Message::BitVec { bit_vec } =>
+                                    (Payload::BitVec(bit_vec), metrics::Message::new_bit_vec(&peer)),
+                                Message::Peers { peers: msg } =>
+                                    (Payload::Peers(msg), metrics::Message::new_peers(&peer)),
                                 Message::Kill => {
                                     return Err(Error::PeerKilled(hex(&peer)))
                                 }
-                            }
+                            };
+                            Self::send_message(&mut conn_sender,payload,metric_msg,&self.sent_messages)
+                                .await?;
                         },
                         msg_high = self.high.next() => {
-                            let msg = match msg_high {
-                                Some(msg_high) => msg_high,
-                                None => return Err(Error::PeerDisconnected),
-                            };
-                            let entry = rate_limits.get(&msg.channel);
-                            if entry.is_none() {
-                                return Err(Error::InvalidChannel);
-                            }
-                            Self::send_content(&mut conn_sender, &peer, msg, &self.sent_messages).await?;
+                            let data = Self::validate_prioritized_data(msg_high, |key| rate_limits.contains_key(key))?;
+                            Self::send_content(&mut conn_sender, &peer, data, &self.sent_messages).await?;
```

Comment:
`send_message` could be used instead of `send_content` here. If yes, we can delete `send_content` function.